plugins/search-backend/src/service/router.ts

```diff
@@ -75,6 +85,8 @@ export async function createRouter(
   const router = await createOpenApiRouter();
   const { engine: inputEngine, types, permissions, config, logger } = options;
 
+  const { auth, httpAuth } = createLegacyAuthAdapters(options);
```

Comment:
Is it a problem that we will implicitly create a `NoopTokenManager`? (as `options` does not contain a `TokenManager`). Just thinking as we are adding `discovery` as a required option, we could also add `tokenManager` and `identity` to ensure we properly validate the token here when those dependencies are available. Or did we actively decide against that in this case?

Reply:
Yep this was indeed troublesome, updates in https://github.com/backstage/backstage/pull/23285. We stopped defaulting to the no-op manager too, but still implement the same behavior